SparseGrids/tsgGridFourier.cpp

```diff
@@ -0,0 +1,574 @@
+        y[i] = y_tmp[i].real();
+    }
+    
+    delete[] y_tmp;
+    delete[] weights;
+    #else
+    evaluateBatch(x, num_x, y);
+    #endif // Tasmanian_ENABLE_BLAS
+}
+
+
+void GridFourier::evaluateBatchGPUcublas(const double x[], int num_x, double y[], std::ostream*) const {
+    evaluateBatchCPUblas(x, num_x, y);
+}
+
+void GridFourier::evaluateBatchGPUcuda(const double x[], int num_x, double y[], std::ostream*) const {
+    evaluateBatchCPUblas(x, num_x, y);
+}
+
+void GridFourier::evaluateBatchGPUmagma(const double x[], int num_x, double y[], std::ostream*) const {
+    evaluateBatchCPUblas(x, num_x, y);
+}
+
+void GridFourier::integrate(double q[], double *conformal_correction) const{
+    double *w = getQuadratureWeights();
+    if(conformal_correction != 0) { for(int i=0; i<points->getNumIndexes(); i++) w[i] *= conformal_correction[i]; }
+    std::fill(q, q+num_outputs, 0.0);
+    #pragma omp parallel for schedule(static)
+    for(int k=0; k<num_outputs; k++) {
+        for(int i=0; i<points->getNumIndexes(); i++) {
+            const double *v = values->getValues(i);
+            q[k] += w[i] * v[k];
+        }
+    }
+    delete[] w;
+}
+
+void GridFourier::evaluateHierarchicalFunctions(const double x[], int num_x, double y[]) const{
+    // y must be of size num_x * num_nodes * 2
+    int num_points = getNumPoints();
+    //#pragma omp parallel for
+    for(int i=0; i<num_x; i++) {
+        getBasisFunctions(&(x[((size_t) i) * ((size_t) num_dimensions)]), &(y[((size_t) i) * ((size_t) 2*num_points)]));
+    }
+}
+
+void GridFourier::setHierarchicalCoefficients(const double c[], TypeAcceleration, std::ostream*) {
+    
+    // takes c to be length 2*num_outputs*num_nodes
+    // first two entries are real and imag parts of the Fourier coef for the first basis function and first output dimension
+    // second two entries are real/imag parts of the Fourier coef for the second basis function and first output dim
+    // and so on
+    if(accel != 0) accel->resetGPULoadedData();
+    for(int i=0; i<num_outputs*getNumPoints(); i++) {
+        fourier_coefs[i] = std::complex<double>(c[2*i], c[2*i+1]);
+    }
+}
+
+void GridFourier::clearAccelerationData() {
+    if(accel != 0) {
+        delete accel;
+        accel = 0;
+    }
+}
+
+void GridFourier::clearRefinement() { return; }     // to be expanded later
+void GridFourier::mergeRefinement() { return; }     // to be expanded later
+
+const int* GridFourier::getPointIndexes() const{
+    return ((points == 0) ? needed->getIndex(0) : points->getIndex(0));
+}
+
+const IndexSet* GridFourier::getExponents() const{
+    return exponents;
+}
+
+const double* GridFourier::getFourierCoefs() const{
+    double* fc = new double[2*getNumPoints()];
+    for(int i=0; i<exponents->getNumIndexes(); i++) {
+        fc[2*i] = fourier_coefs[i].real();
+        fc[2*i+1] = fourier_coefs[i].imag();
+    }
+    return fc;
+}
+
+bool GridFourier::blasEnabled() const {
```

Comment:
* `TasmanianSparseGrids` class has `enabled/disabled` functions, what is the purpose of this function here?